crates/node/src/providers/robust_ecdsa.rs

```diff
@@ -0,0 +1,251 @@
+        // comes directly from generic implementations, so probably this is the best place
+        // to do so anyway
+        let mut old_participants_patched = old_participants.clone();
+        old_participants_patched.threshold = translate_threshold(
+            old_participants.threshold as usize,
+            old_participants.participants.len(),
+        ) as u64;
+
+        EcdsaSignatureProvider::run_key_resharing_client_internal(
+            new_robust_ecdsa_threshold,
+            my_share,
+            public_key,
+            &old_participants_patched,
+            channel,
+        )
+        .await
+    }
+
+    async fn process_channel(&self, channel: NetworkTaskChannel) -> anyhow::Result<()> {
+        match channel.task_id() {
+            MpcTaskId::RobustEcdsaTaskId(task) => match task {
+                RobustEcdsaTaskId::KeyGeneration { .. } => {
+                    anyhow::bail!("Key generation rejected in normal node operation");
+                }
+                RobustEcdsaTaskId::KeyResharing { .. } => {
+                    anyhow::bail!("Key resharing rejected in normal node operation");
+                }
+                RobustEcdsaTaskId::Presignature { id, domain_id } => {
+                    self.run_presignature_generation_follower(channel, id, domain_id)
+                        .await?;
+                }
+                RobustEcdsaTaskId::Signature {
+                    id,
+                    presignature_id,
+                } => {
+                    self.make_signature_follower(channel, id, presignature_id)
+                        .await?;
+                }
+            },
+
+            _ => anyhow::bail!(
+                "eddsa task handler: received unexpected task id: {:?}",
+                channel.task_id()
+            ),
+        }
+        Ok(())
+    }
+
+    async fn spawn_background_tasks(self: Arc<Self>) -> anyhow::Result<()> {
+        let generate_presignatures = self
+            .per_domain_data
+            .iter()
+            .map(|(domain_id, data)| {
+                tracking::spawn(
+                    &format!("generate presignatures for domain {}", domain_id.0),
+                    Self::run_background_presignature_generation(
+                        self.client.clone(),
+                        self.mpc_config.clone(),
+                        self.config.presignature.clone().into(),
+                        *domain_id,
+                        data.presignature_store.clone(),
+                        data.keyshare.clone(),
+                    ),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        for task in generate_presignatures {
+            task.await??;
+        }
+
+        Ok(())
+    }
+}
+
+pub(super) fn get_number_of_signers(threshold: usize, _number_of_participants: usize) -> usize {
+    // TODO: this is the case for the other schemes, might not be our best choice for this one
+    threshold
+}
```

Comment:
This looks incorrect. Shouldn't it be 2t + 1?

Reply:
the inputs refer to the threshold as stored in the contract. As we agreed a few days ago, and as it happens for other schemes, the number of signers equals the threshold is our invariant. The other function that computes the robust ecdsa threshold is the one using the equation you mention